core/app/finders/spree/taxons_finder.rb

```diff
@@ -0,0 +1,59 @@
+module Spree
+  class TaxonsFinder
+    def initialize(taxons, params)
+      @taxons = taxons
+      @params = params
+    end
+
+    def execute
+      taxons = @taxons
+
+      taxons = by_ids(taxons)      if ids?
+      taxons = by_taxonomy(taxons) if taxonomy?
+      taxons = by_roots(taxons)    if roots?
+      taxons = by_name(taxons)     if name?
+
+      taxons
+    end
+
+    private
+
+    attr_accessor :params
+
+    def ids?
+      params[:ids].present?
+    end
+
+    def taxonomy?
+      params[:taxonomy_id].present?
+    end
+
+    def roots?
+      params[:roots].present?
```

Comment:
I don't really think it's finder responsibility to check for the presence of attributes. Can we move it to a separate class? @bbonislawski What do you think?

Reply:
Could you elaborate on that? How would you see these classes implemented and used? Some pseudo-code will be enough